lib/manageiq/floe/workflow/states/pass.rb

```diff
@@ -8,11 +8,29 @@ class Pass < ManageIQ::Floe::Workflow::State
           attr_reader :end, :next, :result, :result_path
 
           def initialize(workflow, name, payload)
+            require "more_core_extensions/core_ext/hash/nested"
+            require "more_core_extensions/core_ext/array/nested"
             super
 
             @next        = payload["Next"]
             @result      = payload["Result"]
-            @result_path = payload["ResultPath"]
+            @result_path = JsonPath.new(payload["ResultPath"]) if payload.key?("ResultPath")
+          end
+
+          def run!
+            logger.info("Running state: [#{name}]")
+
+            if result
+              path = result_path.path[1..]
+                                .map { |v| v.match(/\[(?<name>.+)\]/)["name"] }
+                                .map { |v| v[0] == "'" ? v.gsub("'", "") : v.to_i }
```

Comment:
This is gross and needs refactoring, but this is dealing with turning
```
>> JsonPath.new("$.store.book[0].author").path
=> ["$", "['store']", "['book']", "[0]", "['author']"]
```

Into
`["store", "book", 0, "author"]`

for `Hash#store_path`

Reply:
ok yeah it's temporary, but should suffice for now.